### PR TITLE
Make RNN op generic

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: api/src/main/java/ai/djl/ndarray/internal/NDArrayEx.java [hunk 1]
```diff
@@ -313,67 +314,90 @@ NDList batchNorm(
...
+            NDArray input,
+            NDArray state,
+            NDList params,
+            boolean hasBiases,
+            int numLayers,
+            RNN.Activation activation,
+            double dropRate,
+            boolean training,
+            boolean bidirectional,
+            boolean batchFirst);
```
Comment (roywei) [inline api/src/main/java/ai/djl/ndarray/internal/NDArrayEx.java:344]: `batch_first` seems a confusing flag only for PyTorch, can we only use one behavior? Is there a specific use case for batch not first? 
Ref: https://github.com/pytorch/pytorch/issues/4145

File: api/src/main/java/ai/djl/nn/recurrent/RecurrentBlock.java [hunk 2]
```diff
@@ -297,45 +190,56 @@ public T setStateSize(int stateSize) {
...
-            this.useBidirectional = useBidirectional;
+        public T optHasBiases(boolean hasBiases) {
+            this.hasBiases = hasBiases;
             return self();
         }
 
         /**
-         * Sets the optional parameter that indicates whether to have the states as symbol outputs.
+         * Sets the optional flag that indicates whether to return state or not. This is typically
+         * useful when you use RecurrentBlock in Sequential block.
```
Comment (roywei) [inline api/src/main/java/ai/djl/nn/recurrent/RecurrentBlock.java:236]: provide default value in comments

**Review by roywei — COMMENTED** [review]
Prefer to remove batch first option unless there is a solid usecase. Otherwise LGTM!

**Review by roywei — APPROVED** [review]
After discussion with Jake offline, let's keep batchFirst option for potential speed ups by setting it to `false`, and set default value to `true` for ease of use.

**Review by zachgk — APPROVED** [review]

**Pull request merged**